src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -4595,6 +4595,7 @@ internal static void ProcessPerHeapHistory(TraceLoadedDotNetRuntime proc, GCPerH
             TraceGC _event = GetLastGC(proc);
             if (_event != null)
             {
+                int numGenerations = data.HasCount ? data.Count : 4;
```

Comment:
I would make this a property returned from the parser (ie in ClrTraceEventParser.cs) 'cause this looks quite internal to the event layout. in fact I don't think HasCount should even be a public property.

Reply:
Sounds like a good idea.